core/src/main/java/org/apache/carbondata/core/indexstore/DataMap.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.carbondata.core.indexstore;
+
+import java.util.Comparator;
+import java.util.Iterator;
+
+import org.apache.carbondata.core.indexstore.row.DataMapKey;
+import org.apache.carbondata.core.indexstore.row.DataMapRow;
+import org.apache.carbondata.core.indexstore.schema.DataMapRowSchema;
+
+/**
+ * Interface for adding and retrieving index data.
+ */
+public interface DataMap {
```

Comment:
Let's try to keep this Api minimun. And try to identify Api that can have default implementation which can be class or abstract class. This will encourage more people to contribute more DataMap implementation.